mlflow/metrics/utils/prompts/v1.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, List
+
+from mlflow.metrics.base import (
+    EvaluationExample,
+)
+from mlflow.metrics.utils.prompt_template import (
+    PromptTemplate,
+)
+
+# TODO: Update the default_mode and default_parameters to the correct values post experimentation
+default_model = "openai:/gpt4"
+default_parameters = {
+    "temperature": 0.0,
+    "max_tokens": 100,
+    "top_p": 1.0,
+}
+grading_system_prompt_template = PromptTemplate(
+    """
+Please act as an impartial judge and evaluate the quality of the provided output which
+attempts to produce output for the provided input based on a provided information.
+You'll be given a grading format below which you'll call for each provided information,
+input and provided output to submit your justification and score to compute the {name} of
+the output.
+
+Input:
+{input}
+
+Provided output:
+{output}
+
+{variables}
+
+Metric definition:
+{definition}
+
+Grading criteria:
+{grading_prompt}
+
+{examples}
+
+And you'll need to submit your grading for the {name} of the output,
+using the following in json format:
+Reasoning for {name}: [your step by step reasoning about the {name} of the output]
+Score for {name}: [your score number between 0 to 4 for the {name} of the output]
```

Comment:
@sunishsheth2009 I think we need to specifically tell the LLM what names to use for keys  in the JSON output. If we aren't specific, the LLM may choose different key names, e.g. "reasoning_for_relevance" vs "reasoning for relevance" vs "Reasoning for relevance"